client/components/orderable-list/index.js

```diff
@@ -3,41 +3,123 @@
  * External dependencies
  */
 import React from 'react';
-import classNames from 'classnames';
 import { Button } from '@wordpress/components';
+import classNames from 'classnames';
+import {
+	DndContext,
+	closestCenter,
+	KeyboardSensor,
+	TouchSensor,
+	MouseSensor,
+	useSensor,
+	useSensors,
+} from '@dnd-kit/core';
+import {
+	SortableContext,
+	sortableKeyboardCoordinates,
+	verticalListSortingStrategy,
+	useSortable,
+} from '@dnd-kit/sortable';
+import {
+	restrictToWindowEdges,
+	restrictToVerticalAxis,
+} from '@dnd-kit/modifiers';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
 
-const ListItem = ( { children, className } ) => {
+const ListItem = ( { id, children, className } ) => {
+	const {
+		attributes,
+		listeners,
+		transform,
+		transition,
+		setDraggableNodeRef,
+		setDroppableNodeRef,
+		isDragging,
+	} = useSortable( {
+		id,
+	} );
+
+	const style = {
+		transform: transform
+			? `translate3d(0, ${
+					transform.y ? Math.round( transform.y ) : 0
+			  }px, ${ isDragging ? '100' : '-100' }px) ${
+					isDragging ? 'scale(1.02)' : ''
+			  }`
+			: undefined,
+		transition,
+	};
+
 	return (
-		<li className={ classNames( 'orderable-list__item', className ) }>
-			<Button className="orderable-list__drag-handle" />
+		<li
+			className={ classNames( 'orderable-list__item', className, {
+				'is-dragged': isDragging,
+			} ) }
+			ref={ setDroppableNodeRef }
+			style={ style }
+		>
+			<div className="orderable-list__drag-handle-wrapper">
+				<Button
+					className="orderable-list__drag-handle"
+					{ ...attributes }
+					{ ...listeners }
+					ref={ setDraggableNodeRef }
+				/>
+			</div>
 			{ children }
 		</li>
 	);
 };
 
-const OrderableList = ( { className, children } ) => {
-	const childrenArray = React.Children.toArray( children );
```

Comment:
Converting the `children` to an array via `React.Children.toArray`, changes the `key` attribute on the child. So, for example, if the child had key with value `'a'`, after the `React.Children.toArray` transformation, the attribute key would have value `$.a` (see note in https://reactjs.org/docs/react-api.html#reactchildrentoarray ).

By leveraging `React.Children.map` instead, the key's value is preserved.